src/Tasks/BootstrapperUtil/ResourceUpdater.cs

```diff
@@ -100,7 +100,14 @@ public bool UpdateResources(string filename, BuildResults results)
                                 fileLength = (int)fs.Length;
                                 fileContent = new byte[fileLength];
 
+#if NET7_0_OR_GREATER
+                                fs.ReadExactly(fileContent, 0, fileLength);
+#else
+#pragma warning disable CA2022 // Avoid inexact read with 'Stream.Read'
+                                // TODO: Read the count of read bytes and check if it matches the expected length, if not raise an exception
                                 fs.Read(fileContent, 0, fileLength);
+#pragma warning restore CA2022 // Avoid inexact read with 'Stream.Read'
+#endif
```

Comment:
I still really want a polyfill for this so we don't have to have details in more than one place.

Reply:
I have added the extensions for the stream reading for the cases where the all file is read. All other cases where only first 2 bytes are read, the warning was disabled, since the validation of first 2 bytes happens on later stages.